### PR TITLE
[SLM] Add unit tests for SLM to Relax exporter

### PR DESCRIPTION
**Pull request by Lunderberg**
Follow-up to https://github.com/apache/tvm/pull/16777, add unit tests demonstrating desired behavior.

### CODE REVIEW

File: tests/python/relax/test_frontend_nn_exporter.py [hunk 1]
```diff
@@ -0,0 +1,632 @@
...
+                relu = R.nn.relu(x)
+                relu = relu
+                R.output(relu)
+            return relu
+
+    assert_structural_equal(exported_mod, Expected)
+
+
+def test_custom_module():
+    """A module may be exported from nn.Module to Relax"""
```
Comment (slyubomirsky) [inline tests/python/relax/test_frontend_nn_exporter.py:53]: Perhaps the comment should differ in soem way from that in the above test.

**Review by slyubomirsky — APPROVED** [review]
It's good to have test cases, especially when they explain the intended functionality very well. A tutorial featuring based on these examples (perhaps literally generated from the test cases) might be a good investment of time as well, especially if it can be made more visible.

File: tests/python/relax/test_frontend_nn_exporter.py (hunk 1 above)
Comment (Lunderberg) [reply tests/python/relax/test_frontend_nn_exporter.py]: Good call, and I've updated the docstring to remove the copy/paste duplicate.

**Comment by Lunderberg** [comment]
> It's good to have test cases, especially when they explain the intended functionality very well. A tutorial featuring based on these examples (perhaps literally generated from the test cases) might be a good investment of time as well, especially if it can be made more visible.

Thank you, and that was in part my goal with the sequence of unit tests.  Whenever possible, I prefer test cases that double as mini-tutorials, since then they are less likely to become stale than full tutorials.  (Though I agree that it would be beneficial to have a tutorial that follows user-focused flow, rather than the feature-focused flow of these unit tests.)

**Review by MasterJH5574 — APPROVED** [review]
Thank you @Lunderberg 🙏

**Pull request merged**